src/core_codemods/upgrade_sslcontext_minimum_version.py

```diff
@@ -45,6 +48,11 @@ def rule(cls):
         """
 
     def on_result_found(self, original_node, updated_node):
+        maybe_name = self.get_aliased_prefix_name(
+            original_node.value, self._module_name
+        )
+        maybe_name = maybe_name or self._module_name
+        if maybe_name and maybe_name == self._module_name:
+            self.add_needed_import(self._module_name)
```

Comment:
these new green lines added are the same (or almost?) in every codemod added so maybe make this a separate method to call to dedupe

Reply:
These address a very specific problem that are shared by those codemods. I don't want to add it to the api because the solution needs `NameResolutionMixin` which depends on `ScopeProvider`.

Metadata is calculated at the start of every transform on a as-need basis (libcst looks at `METADATA_DEPENDENCIES` for that). Integrating it to the API means every codemod would calculate the `ScopeProvider` metadata, making it a bit more expensive.

Maybe creating a class for those types of codemods would be a good idea, but I'd rather do that when we try to rewrite those in pure libcst.

Reply:
I agree the duplication here is not ideal. I think we need to revisit the whole concept of utility mixins for this reason, although I understand the reasons it was done this way.